src/yang/ietf-access-control-list.yang

```diff
@@ -422,145 +457,132 @@ module ietf-access-control-list {
+              if-feature match-on-ipv6;
               uses packet-fields:acl-ip-header-fields;
               uses packet-fields:acl-ipv6-header-fields;
               description
                 "Rule set that supports IPv6 headers.";
             }
 
-            container l2-l3-ipv4-acl {
-              if-feature mixed-ipv4-acl;
-              must "derived-from(../../../../acl-type, " +
-                   "'acl:mixed-l2-l3-ipv4-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv4-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of l2 
-                 and ipv4 headers.";
-            }
-
-            container l2-l3-ipv6-acl {
-              if-feature mixed-ipv6-acl;
-              must "derived-from(../../../../acl-type, " +
-                   "'acl:mixed-l2-l3-ipv6-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv6-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of L2 
-                 && IPv6 headers.";
-            }
-
-            container l2-l3-ipv4-ipv6-acl {
-              if-feature l2-l3-ipv4-ipv6-acl;
-              must "derived-from(../../../../acl-type, " +
-                   "'acl:mixed-l2-l3-ipv4-ipv6-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv4-header-fields;
-              uses packet-fields:acl-ipv6-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of L2 
-                 && IPv4 && IPv6 headers.";
-            }
-            
-            container tcp-acl {
-              if-feature tcp-acl;
+            container tcp {
+              if-feature match-on-tcp;
               uses packet-fields:acl-tcp-header-fields;
               description
                 "Rule set that defines TCP headers.";
             }
-            
-            container udp-acl {
-              if-feature udp-acl;
+
+            container udp {
+              if-feature match-on-udp;
               uses packet-fields:acl-udp-header-fields;
               description
                 "Rule set that defines UDP headers.";
             }
-            
-            container icmp-acl {
-              if-feature icmp-acl;
+
+            container icmp {
+              if-feature match-on-icmp;
               uses packet-fields:acl-icmp-header-fields;
               description
                 "Rule set that defines ICMP headers.";
             }
-             
-            container any-acl {
-              if-feature any-acl;
-              must "derived-from(../../../../acl-type, 'acl:any-acl')";
-              presence "Matches any";
+
+            leaf any {
```

Comment:
Remove the any-acl altogether. It serves no purpose. We can match on anything by having an ACE with no match conditions at all - only an action.